### PR TITLE
[v2.0] Move exchange keys into User

### PR DESCRIPTION
**Pull request by DyrellC**
# Description of change
Remove exchange key management from `KeyStore` and rename to `CursorStore`. Exchange keys will now be hosted inside the `User` for now (will be removed later once #245 is addressed). 

Addresses: #240 

## Type of change
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested
- examples and tests continue to run with changes
- cargo fmt/clippy/test

### CODE REVIEW

**Review by arnauorriols — APPROVED** [review]
All good! I would say something about the method naming of `CursorStore`, but I assume they will change again the topic PR

File: streams/src/api/user.rs [hunk 1]
```diff
@@ -854,18 +864,18 @@ impl<'a> ContentWrap<State> for wrap::Context<&'a mut [u8]> {
...
-        let cursors = user_state.id_store.cursors();
+        let cursors = user_state.cursor_store.cursors();
         let amount_cursors = cursors.len();
         self.mask(Size::new(amount_cursors))?;
         for (subscriber, cursor) in cursors {
             self.mask(&subscriber)?.mask(Size::new(cursor))?;
         }
 
-        let keys = user_state.id_store.keys();
+        let keys = user_state.exchange_keys.iter().map(|(id, xkey)| (id, xkey));
```
Comment (kwek20) [inline streams/src/api/user.rs:874]: same for here

**Review by kwek20 — CHANGES_REQUESTED** [review]
lets clean this,  then good to go!

File: streams/src/api/user.rs [hunk 2]
```diff
@@ -816,18 +826,18 @@ impl ContentSizeof<State> for sizeof::Context {
...
```
[the remark this answers is not in the document]
Comment (DyrellC) [reply streams/src/api/user.rs:836]: Yes, yes we can

File: streams/src/api/user.rs (hunk 1 above)
Comment (DyrellC) [reply streams/src/api/user.rs]: done

**Pull request merged**